src/shared/listener-collection.js

```diff
@@ -11,24 +11,41 @@
  */
 export class ListenerCollection {
   constructor() {
-    /** @type {Listener[]} */
+    /** @type {Array<Listener|null>} */
     this._listeners = [];
   }
 
   /**
+   * Adds a listener and returns a listenerId
    * @param {Listener['eventTarget']} eventTarget
    * @param {Listener['eventType']} eventType
    * @param {Listener['listener']} listener
    * @param {AddEventListenerOptions} [options]
    */
   add(eventTarget, eventType, listener, options) {
     eventTarget.addEventListener(eventType, listener, options);
-    this._listeners.push({ eventTarget, eventType, listener });
+    return this._listeners.push({ eventTarget, eventType, listener }) - 1;
```

Comment:
The first listener ID is returned is 0, which is a falsey value. This could cause problems if calling code tries to do `if (!listenerId) { ... }`. I think it would be safer to make IDs start from 1, like `setTimeout` IDs.